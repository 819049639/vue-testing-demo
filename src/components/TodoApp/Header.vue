<!--
 * @Author: wangqiang@feewee.cn
 * @Date: 2022-07-19 14:44:04
 * @LastEditors: wangqiang@feewee.cn
 * @LastEditTime: 2022-07-19 14:59:21
-->
<template>
  <header class="header">
    <h1>todos</h1>
    <input
      class="new-todo"
      data-testid="new-todo"
      placeholder="What needs to be done?"
      autofocus
      @keyup.enter="handleNewTodo"
    />
  </header>
</template>
<script>
export default {
  name: 'TodoAppHeader',
  methods: {
    handleNewTodo (e) {
      const value = e.target.value.trim()
      if (!value) {
        return
      }
      this.$emit('new-todo', value)
      e.target.value = ''
    }
  }
}
</script>
