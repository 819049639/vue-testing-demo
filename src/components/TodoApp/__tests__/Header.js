/*
 * @Author: wangqiang@feewee.cn
 * @Date: 2022-07-19 14:50:17
 * @LastEditors: wangqiang@feewee.cn
 * @LastEditTime: 2022-07-20 09:25:11
 */
import { shallowMount } from '@vue/test-utils'
import Header from '../Header.vue'

describe('Header.vue', () => {
  /** @type {import('@vue/test-utils').Wrapper} */
  let wrapper = null

  beforeEach(async () => {
    wrapper = shallowMount(Header)
  })

  test('New Todo', async () => {
    const input = wrapper.find('input[data-testid="new-todo"]')

    const text = 'play'
    await input.setValue(text)

    await input.trigger('keyup.enter')
    expect(wrapper.emitted()['new-todo']).toBeTruthy()
    expect(wrapper.emitted()['new-todo'][0][0]).toBe(text)
    expect(input.element.value).toBe('')
  })

  test('snapshot', async () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
