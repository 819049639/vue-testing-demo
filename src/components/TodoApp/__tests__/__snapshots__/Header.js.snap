// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.vue snapshot 1`] = `
<header class="header">
  <h1>todos</h1> <input data-testid="new-todo" placeholder="What needs to be done?" autofocus="autofocus" class="new-todo">
</header>
`;
